src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -962,7 +962,8 @@ namespace AppInstaller::CLI::Workflow
         bool isUpdate = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseUpdate);
 
         IPackageVersion::Metadata installationMetadata;
-        Utility::Architecture requiredArchitecture = Settings::User().Get<Settings::Setting::InstallArchitectureRequirement>();
+        std::vector<Utility::Architecture> requiredArchitectures = Settings::User().Get<Settings::Setting::InstallArchitectureRequirement>();

```

Comment:
I meant that these settings should only be retrieved in the `else` that needs to read them, thus preventing the work to get them if an argument was given.  Minor optimization, but also an easy one.  I'll stop commenting on it though, so up to you.

Reply:
I moved it, that makes more sense.